internal/pkg/s3/s3.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package s3 provides a client to retrieve Copilot S3 information.
+package s3
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/copilot-cli/internal/pkg/aws/resourcegroups"
+	"github.com/aws/copilot-cli/internal/pkg/aws/s3"
+	"github.com/aws/copilot-cli/internal/pkg/deploy"
+)
+
+const (
+	bucketType = "s3:bucket"
+)
+
+type resourceGetter interface {
+	GetResourcesByTags(resourceType string, tags map[string]string) ([]*resourcegroups.Resource, error)
+}
+
+// Client retrieves Copilot S3 service information from AWS.
+type Client struct {
+	rgGetter resourceGetter
+}
+
+// New inits a new Client.
+func New(sess *session.Session) *Client {
+	return &Client{
+		rgGetter: resourcegroups.New(sess),
+	}
+}
+
+// BucketName returns the bucket name given the Copilot app, env, and Static Site service name.
+func (c Client) BucketName(app, env, svc string) (string, error) {
+	buckets, err := c.rgGetter.GetResourcesByTags(bucketType, map[string]string{
+		deploy.AppTagKey:     app,
+		deploy.EnvTagKey:     env,
+		deploy.ServiceTagKey: svc,
+	})
+	if err != nil {
+		return "", fmt.Errorf("get S3 bucket with tags (%s, %s, %s): %w", app, env, svc, err)
+	}
+	if len(buckets) == 0 {
+		return "", fmt.Errorf("no S3 bucket found with tags %s, %s, %s", svc, env, svc)
```

Comment:
The second error message also has `svc` twice instead of `app` 😅

Reply:
Lol I'll fix in the current PR.

> and invoking tags.String() in the error messages

We can probably add it if it happens more in this lib, since it also takes extra logic to construct a map that's only used for err msg.

Reply:
sounds good! but it'd be helpful either way to also include the keys of the tags and not just the values so that customers can debug

Reply:
Actually I'll update with `tags`. It's a great idea!